docs/make.md

```diff
@@ -0,0 +1,186 @@
+
+Create a file called `Makefile` and copy-paste this code.
+```make
+CXX ?= g++
+CXXFLAGS ?= -Wall -Werror -pedantic -g --std=c++11 -Wno-sign-compare -Wno-comment
+
+# Run regression test
+test: main.exe
+	./main.exe < main_test.in > main_test.out
+	diff main_test.out main_test.out.correct
+	echo PASS
+
+# Compile the main executable
+main.exe: main.cpp
+	$(CXX) $(CXXFLAGS) main.cpp -o main.exe
+
+# Remove automatically generated files
+clean :
+	rm -rvf *.exe *~ *.out *.dSYM *.stackdump
+```
+{: data-title="Makefile" }
+
+Your directory should look like this:
+```console
+$ tree
+.
+├── Makefile
+└── main.cpp
+```
+
+
+## Compile
+Compile manually, then remove the executable.
+```console
+$ g++ -Wall -Werror -pedantic -g --std=c++11 -Wno-sign-compare -Wno-comment main.cpp -o main.exe
+$ ./main.exe
+Hello World!
+$ rm main.exe
+```
+
+Compile using `make`.  Notice that it automated the long compile command.
+```console
+$ make main.exe
+g++ -Wall -Werror -pedantic -g --std=c++11 -Wno-sign-compare -Wno-comment main.cpp -o main.exe
+```
+
+Run the executable created by the previous `make` command.
+```console
+$ ./main.exe
+Hello World!
+```
+
+Clean up using `make`.  Notice that it automated the `rm` command.
+```console
+$ make clean
+rm -rvf *.exe *~ *.out *.dSYM *.stackdump
+removed 'main.exe'
+```
+
+
+## Test
+Makefiles are useful for automating tests, sometimes called a *regression test*.
+
+Put input (stdin) in a file.  We'll use `main_test.in` in this example.
+```
+this program doesn't read stdin,
+so this text doesn't matter
+```
+{: data-title="main_test.in" }
+
+Put the correct output in a file.  We'll use `main_test.out.correct`.  In EECS 280 and EECS 281, a few correct output files are typically provided.
+```
+Hello World!
+```
+{: data-title="main_test.out.correct" }
+
+Run manually with [input redirection](cli.html#input-redirection-) and [output redirection](cli.html#output-redirection-).  Compare with [`diff`](cli.html#diff).  No output means the files match.
+```console
+$ ./main.exe < main_test.in > main_test.out
+$ diff main_test.out main_test.out.correct
+```
+
+Run the regression test automatically with `make`.
+```console
+$ make test
+./main.exe < main_test.in > main_test.out
+diff main_test.out main_test.out.correct
+```
+
+<div class="primer-spec-callout info" markdown="1">
+**Pro-tip:** Run commands in parallel with `make -j`.
+```console
+$ make -j4 test
+```
+{: data-variant="no-line-numbers" }
+</div>
+
+
+## Makefile syntax
+This section explains basic `Makefile` syntax.
+
+A `Makefile` consists of *rules* that look like this:
+```make
+target: prerequisites
+	recipe
+```
+
+This is a concrete example of a rule.
+```make
+main.exe: main.cpp
+	g++ main.cpp -o main.exe
+```
+
+A *target* is usually the name of a file that is created by the rule, like `main.exe`.
+
+A *prerequisite* is a file that is used as input to create the target, like `main.cpp`.  There could be multiple prerequisites.
```

Comment:
Perhaps a line that mentions that .cpp files should be used here, not .h files. This could be confusing at first.

Reply:
I don't think I understand.  Here's two different examples that are pretty common:

```make
stats.cpp : stats.h

main.exe : main.cpp
  g++ main.cpp -o main.exe
```

Reply:
Oh perhaps I don't understand Makefiles well enough...